lib/DxcSupport/dxcmem.cpp

```diff
@@ -43,7 +43,7 @@ HRESULT DxcInitThreadMalloc() throw() {
 void DxcCleanupThreadMalloc() throw() {
   if (g_ThreadMallocTls) {
     DXASSERT(g_pDefaultMalloc, "else DxcInitThreadMalloc didn't work/fail atomically");
-    g_ThreadMallocTls->llvm::sys::ThreadLocal<IMalloc>::~ThreadLocal();
+    g_ThreadMallocTls->llvm::sys::ThreadLocal<IMalloc>::~ThreadLocal<IMalloc>();
```

Comment:
@jaebaek This warning shows up on master regardless, and has been fixed by removing the `ThreadLocal` classification entirely in https://github.com/microsoft/DirectXShaderCompiler/pull/3794/commits/43d09822e571b8d8d56c8d0ed20f8906f63268bf. Inclined to remove it from this PR, do you think that PR should use the full `llvm::sys::ThreadLocal<IMalloc>::~ThreadLocal<IMalloc>` name instead?

Reply:
Oh, I did not know you prepared another commit for it. With clang on Linux, `g_ThreadMallocTls->~ThreadLocal()` also works fine.